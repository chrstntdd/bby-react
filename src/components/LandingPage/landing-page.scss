.hero-img {
  background-image: url('images/hero-bg-1.jpg');
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-attachment: scroll;
}
