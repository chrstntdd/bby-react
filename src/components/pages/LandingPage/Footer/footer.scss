@import '$homeDir/styles/rwd-helpers.scss';
@import '$homeDir/styles/colors.scss';

footer {
  width: 100vw;
  background-color: $BESTBUY_BLUE;
  padding: auto 50px;
  ul {
    padding-top: 1%;
    padding-left: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    margin: 0 auto;
    li {
      @media (min-width: $bp-xxsmall) {
        margin-top: between(9px, 12px, $bp-xxsmall, $bp-xlarge);
        margin-bottom: between(9px, 12px, $bp-xxsmall, $bp-xlarge);
      }
      list-style: none;
      padding-left: 0;
      a {
        letter-spacing: 2px;
        text-transform: uppercase;
        font-size: 1rem;
        text-decoration: none;
        color: white;
        transition-property: color;
        transition-duration: 0.5s;
        &:hover {
          transition-property: color;
          color: $BESTBUY_YELLOW;
        }
      }
    }
  }
}
