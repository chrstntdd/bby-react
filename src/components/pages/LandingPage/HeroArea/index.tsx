import * as React from 'react';
import { Link } from 'react-router-dom';

import './hero-area.scss';

const HeroArea = () => (
  <section id="hero-atf">
    <div className="button-container">
      <Link id="sign-up-button" to="/sign-up">
        <button id="btn-signup">Sign Up</button>
      </Link>

      <Link id="sign-in-button" to="/sign-in">
        <button id="btn-signin">Sign In</button>
      </Link>
    </div>
    <div className="hero-img" />
    <div className="hero-content">
      <h1>Quantified</h1>
      <h2>A streamlined system for efficiently and accurately managing retail inventory.</h2>
    </div>
  </section>
);

export default HeroArea;
