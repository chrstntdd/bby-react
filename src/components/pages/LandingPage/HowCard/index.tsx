import * as React from 'react';
import './how-card.scss';

const Card = props => (
  <div className="how-card">
    <img src={props.img} alt={props.name + ' icon'} />
    <h3>{props.name}</h3>
    <h5>{props.desc}</h5>
  </div>
);

export default Card;
