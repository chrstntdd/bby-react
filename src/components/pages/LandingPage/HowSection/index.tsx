import * as React from 'react';
import './how-section.scss';
import Card from '../HowCard';

const HowSection = props => {
  const howCards = props.cardData.map((card, index) => <Card key={index} {...card} />);
  return (
    <section id="how-section">
      <div className="card-container">{howCards}</div>
    </section>
  );
};

export default HowSection;
