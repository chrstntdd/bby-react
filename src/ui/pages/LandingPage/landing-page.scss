@import '$homeDir/styles/colors.scss';

.hero-img {
  background-image: url('images/hero-bg-1.jpg');
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-attachment: scroll;
}

.lp-max-width {
  max-width: 1024px;
}

.about-copy {
  margin: 2rem auto;
}

.how-card-cont {
  border-top: 0.4rem solid $BESTBUY_YELLOW;
  height: 20rem;
  width: 13rem;
  background-color: hsl(200, 25%, 88%);
  box-shadow: 0 16px 18px 0 rgba(0, 0, 0, 0.13), 0 4px 6px 0 rgba(0, 0, 0, 0.16);
}

.ofc {
  object-fit: cover;
}
