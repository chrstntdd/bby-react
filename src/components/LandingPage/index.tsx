import * as React from 'react';
import { Link } from 'react-router-dom';

import './landing-page.scss';

const navigationButtons =
  'trans-300ms-all z-10 font-semibold rounded-full px-8 py-2 leading-normal bg-transparent border border-grey text-grey hover:border-bby-blue hover:bg-bby-blue hover:text-white m-6';

const LandingPage = () => (
  <div>
    <section className="h-screen w-screen">
      <div className="w-full absolute pin-t z-10 flex justify-center md:justify-end">
        <Link to="/sign-up">
          <button className={navigationButtons}>Sign Up</button>
        </Link>

        <Link to="/sign-in">
          <button className={navigationButtons}>Sign In</button>
        </Link>
      </div>
      <div className="hero-img h-screen w-screen absolute pin" />
      <div className="kinda-center z-10 w-full">
        <h1
          className={
            'text-white italic text-center leading-none whitespace-no-wrap text-5xl md:text-massive tracking-wide mb-2'
          }
        >
          Quantified
        </h1>
        <h2 className="text-white italic text-center leading-none text-xl font-thin">
          A streamlined system for efficiently and accurately managing retail inventory.
        </h2>
      </div>
    </section>
    <section id="how-section">
      <div className="card-container">
        <div className="how-card">
          <img src={'images/atom.svg'} alt={'atom icon'} />
          <h3>Simple</h3>
          <h5>...but powerful</h5>
        </div>
        <div className="how-card">
          <img src={'images/handshake.svg'} alt={'handshake icon'} />
          <h3>Reliable</h3>
          <h5>There when you need it</h5>
        </div>
        <div className="how-card">
          <img src={'images/padlock.svg'} alt={'padlock icon'} />
          <h3>Safe</h3>
          <h5>Without compromises</h5>
        </div>
        <div className="how-card">
          <img src={'images/rocket.svg'} alt={'rocket icon'} />
          <h3>Fast</h3>
          <h5>Designed with speed in mind</h5>
        </div>
      </div>
    </section>
    <section id="about-section">
      <h1>About Quantified</h1>
      <blockquote>
        "A streamlined system for efficiently and accurately managing retail inventory."
      </blockquote>
      <p>
        That sounds great and all, but what does that mean exactly? Imagine for a moment that you
        are in a retail setting. When shipments of products come into your store, they will be
        accompanied by a manifest that notates exactly what products and how many of them were put
        onto the trailer when it left the distribution center. It is the job of one employee to
        verify everything listed on the manifest actually makes it to the store, not an easy or
        quick task by any means.
      </p>
      <p>
        The previous workflow (at least at Best Buy) for counting all of these products involved
        printing out a copy of the aforementioned manifest, then tallying up the products one by
        one. Sounds monotonous and error prone right? That's where Quantified comes in to save the
        day by bearing the responsibility of accounting for everything.
      </p>
      <p>
        Instead of manually counting each product, Quantified allows the user to simply scan the bar
        code of the product and automatically populates all of its relevant information into a row
        in a table. From there, after everything has been scanned, the table can be formatted to
        mirror the manifest and printed for seamless comparison with the manifest for the final
        verification.
      </p>
      <p>
        Additionally, Quantified is built with battle tested and industry leading technology as well
        as the Best Buy API. All together this yields unmatched dependability, speed, and security.
      </p>
    </section>
    <footer>
      <ul>
        <li>
          <Link to={'/sign-up'}>Start here</Link>
        </li>

        <li>
          <Link to={'/sign-in'}>Sign in</Link>
        </li>

        <li>
          <a href={'https://developer.bestbuy.com/api-status'} target="_blank">
            Official Best Buy API Status
          </a>
        </li>

        <li>
          <a href="https://github.com/chrstntdd" target="_blank">
            Created by Christian Todd
          </a>
        </li>
      </ul>
    </footer>
  </div>
);

export default LandingPage;
