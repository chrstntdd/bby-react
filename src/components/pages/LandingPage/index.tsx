import * as React from 'react';

import HeroArea from './HeroArea';
import HowSection from './HowSection';
import AboutSection from './AboutSection';
import Footer from './footer';

const simple = 'images/atom.svg';
const reliable = 'images/handshake.svg';
const secure = 'images/padlock.svg';
const fast = 'images/rocket.svg';

export default class LandingPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      footerData: [
        {
          name: 'Start here',
          link: '/sign-up'
        },
        {
          name: 'Sign in',
          link: '/sign-in'
        },
        {
          name: 'Official Best Buy API Status',
          link: 'https://developer.bestbuy.com/api-status'
        }
      ],
      cardData: [
        {
          name: 'simple',
          img: simple,
          desc: '...but powerful'
        },
        {
          name: 'reliable',
          img: reliable,
          desc: 'There when you need it'
        },
        {
          name: 'safe',
          img: secure,
          desc: 'Without compromises'
        },
        {
          name: 'fast',
          img: fast,
          desc: 'Designed with speed in mind'
        }
      ]
    };
  }
  render() {
    return (
      <div className="landing-page">
        <HeroArea />
        <HowSection cardData={this.state.cardData} />
        <AboutSection />
        <Footer footerData={this.state.footerData} />
      </div>
    );
  }
}
