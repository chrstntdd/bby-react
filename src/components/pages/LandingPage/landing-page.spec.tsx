import * as React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import createRouterContext from 'react-router-test-context';
import * as configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';

import HeroArea from './HeroArea';
import HowSection from './HowSection';
import AboutSection from './AboutSection';
import Footer from './Footer';
import LandingPage from './';

const setup = () => {
  const mockStore = configureStore();
  const context = createRouterContext();
  const initialState = {
    table: {
      products: [],
      formatted: false,
      printing: false,
      tableId: '',
      showModal: false,
      selectOptionData: [
        {
          tableId: '599aec4c2ea3f20011051fe1',
          formattedDate: 'Mon Aug 21 2017-10:21:00 AM'
        }
      ]
    }
  };

  const props = {
    handleShowModal: jest.fn(),
    selectOptionData: [
      {
        tableId: '599aec4c2ea3f20011051fe1',
        formattedDate: 'Mon Aug 21 2017-10:21:00 AM'
      }
    ]
  };
  const store = mockStore(initialState);
  const enzymeWrapper = mount(
    <Router context={context}>
      <Provider store={store}>
        <LandingPage {...props} store={store} />
      </Provider>
    </Router>
  );

  return {
    props,
    enzymeWrapper
  };
};

describe('<LandingPage/>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<LandingPage />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should have state', () => {
    const wrapper = shallow(<LandingPage />);
    const state = wrapper.state();
    expect(state).not.toBe(null);
  });
});
