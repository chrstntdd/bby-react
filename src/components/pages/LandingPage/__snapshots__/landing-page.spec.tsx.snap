// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage/> should render without crashing 1`] = `
<div
  className="landing-page"
>
  <HeroArea />
  <HowSection
    cardData={
      Array [
        Object {
          "desc": "...but powerful",
          "img": "images/atom.svg",
          "name": "simple",
        },
        Object {
          "desc": "There when you need it",
          "img": "images/handshake.svg",
          "name": "reliable",
        },
        Object {
          "desc": "Without compromises",
          "img": "images/padlock.svg",
          "name": "safe",
        },
        Object {
          "desc": "Designed with speed in mind",
          "img": "images/rocket.svg",
          "name": "fast",
        },
      ]
    }
  />
  <AboutSection />
  <Footer
    footerData={
      Array [
        Object {
          "link": "/sign-up",
          "name": "Start here",
        },
        Object {
          "link": "/sign-in",
          "name": "Sign in",
        },
        Object {
          "link": "https://developer.bestbuy.com/api-status",
          "name": "Official Best Buy API Status",
        },
      ]
    }
  />
</div>
`;
